src/Money/Internal.hs

```diff
@@ -1675,35 +1673,50 @@ storeContramapSize f = \case
 -- decimal string due to a bad choice of separators. That is, if the separators
 -- are digits or equal among themselves, this function returns 'Nothing'.
 denseToDecimal
-  :: GoodScale scale
-  => Approximation
+  :: Approximation
   -- ^ Approximation to use if necesary in order to fit the 'Dense' amount in
   -- as many decimal numbers as requested.
   -> Bool
   -- ^ Whether to render a leading @\'+\'@ sign in case the amount is positive.
   -> Maybe Char
   -- ^ Thousands separator for the integer part, if any (i.e., the @\',\'@ in
   -- @1,234.56789@).
+  --
+  -- If the given separator is a digit, or if it is equal to the decimal
+  -- separator, then this functions returns 'Nothing'.
   -> Char
-  -- ^ Decimal separator (i.e., the @\'.\'@ in @1,234.56789@)
+  -- ^ Decimal separator (i.e., the @\'.\'@ in @1,234.56789@).
+  --
+  -- If the given separator is a digit, or if it is equal to the thousands
+  -- separator, then this functions returns 'Nothing'.
   -> Word8
   -- ^ Number of decimal numbers to render, if any.
-  -> Proxy scale
-  -- ^ Scale used by the integer part of the decimal number. For example, a when
-  -- rendering render @'dense'' (123 '%' 100) :: 'Dense' "USD"@ as a decimal
-  -- number with three decimal places, a scale of @1@ (i.e. @'Scale' \"USD\"
-  -- \"dollar\"@) would render @1@ as the integer part and @230@ as the
-  -- fractional part, whereas a scale of @100@ (i.e., @'Scale' \"USD\"
-  -- \"cent\"@) would render @123@ as the integer part and @000@ as the
+  -> Rational
+  -- ^ Scale used by the integer part of the decimal number. This is useful if
+  -- you want to render a “number of cents” rather than a “number of dollars”,
+  -- for example. Set this to @1@ if you don't care.
+  --
+  -- For example, when rendering render @'dense'' (123 '%' 100) :: 'Dense'
+  -- "USD"@ as a decimal number with three decimal places, a scale of @1@ (i.e.
+  -- @'Scale' \"USD\" \"dollar\"@) would render @1@ as the integer part and
```

Comment:
I think the comments for this argument need to be updated to reflect the change to `Rational`?